lib/ethif/ethif.ml

```diff
@@ -42,12 +42,11 @@ module Make(Netif : V1_LWT.NETWORK) = struct
   let mac t = Netif.mac t.netif
 
   let input ~arpv4 ~ipv4 ~ipv6 t frame =
-    let open Ethif_parse in
     MProf.Trace.label "ethif.input";
     let of_interest dest =
       Macaddr.compare dest (mac t) = 0 || not (Macaddr.is_unicast dest)
     in
-    match parse_ethernet_header frame with
+    match Ethif_unmarshal.of_cstruct frame with
```

Comment:
Hm-- to some extent I'm inclined to go with this one and then think further about naming of things. Splitting out the modules seems a good thing regardless of more name wrangling required. And possibly having things split into smaller modules will make it easier to think coherently about what the API should be...

(FWIW I had though that `marshal`/`unmarshal` were to be function names inside something like `Tcp4_wire` rather than embedded directly in module names -- apologies for any noise contributed thereby.)